### ISSUE TITLE
Payment Method Not Showing Admin Sales Grid only check method showing.

### ISSUE THREAD

**Issue by pushpendra07** [magento/magento2#22231]
**Payment Method showing blank always except check method**

### Preconditions (*)

1. Magento version 2.3.1 

### Steps to reproduce (*)

1. Create a simple order with bank transfer or COD.
2. Check Admin Sales order grid

### Expected result (*)

1. Show Payment method in grid

### Actual result (*)

1.Only Check method showing in the list ![checkout1](https://user-images.githubusercontent.com/8010768/55726617-d419ce80-5a2d-11e9-977c-0f3f3a642601.png)
2. Order Edit window showing payment details ![checkout2](https://user-images.githubusercontent.com/8010768/55773193-26e69b00-5aad-11e9-9086-b86ffc07cafd.png)

**Comment by krishprakash** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @krishprakash. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @krishprakash, here is your [Magento instance](https://i-22231-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-22231-2-3-develop.instances.magento-community.engineering/admin](https://i-22231-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by krishprakash** [comment]
![payment method grid](https://user-images.githubusercontent.com/46617848/55774451-c6f2f300-5ab2-11e9-813e-f51d0fd692ce.png)

**Comment by krishprakash** [comment]
@magento-engcom-team give me 2.2-develop instance

**Comment by magento-engcom-team** [comment]
Hi @krishprakash. Thank you for your request. I'm working on Magento 2.2-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @krishprakash, here is your [Magento instance](https://i-22231-2-2-develop.instances.magento-community.engineering).
Admin access: [https://i-22231-2-2-develop.instances.magento-community.engineering/admin](https://i-22231-2-2-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by krishprakash** [comment]
![2 2 x payment grid](https://user-images.githubusercontent.com/46617848/55775064-5699a100-5ab5-11e9-8513-511cfd53a8f5.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @shikhamis11 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-99111`, `MAGETWO-99112` were created

**Issue Available**: @shikhamis11, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by amelojunior** [comment]
temporary fix...

add to etc/config.xml:

    <default>
        <payment>
            <cashondelivery>
                <active>1</active>
            </cashondelivery>
            <banktransfer>
                <active>1</active>
            </banktransfer>
        </payment>
    </default>

**Comment by domeglic** [comment]
The problem seems to be a [new condition](https://github.com/magento/magento2/commit/28c86bed310311247604caa6c5c3b6c165fbd4f3#diff-8256f35d1118cefe17ef3731bbc346abR264) that checks for active payments, but somehow it returns wrong results because in my case the missing methods are active.

**Comment by Julen10** [comment]
@domeglic So, what´s a possible solution for this one?

**Comment by domeglic** [comment]
@Julen10  No idea what a proper solution is, I didn't look into why this additional check was added. The way I solved this for my project is to revert the change only for the sales grid:

Create a class that extends `Magento/Payment/Helper/Data` and removes the active check. Then set it as the argument for the column renderer:
```
<type name="Magento\Payment\Ui\Component\Listing\Column\Method\Options">
        <arguments>
            <argument name="paymentHelper" xsi:type="object">Vendor\Module\Bugfix\Payment\HelperData</argument>
        </arguments>
    </type>
```

**Comment by magento-engcom-team** [comment]
Hi @pushpendra07, @Nazar65, @krishprakash. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.2-develop` branch(es).
Related commit(s):
 - [08ecfe1efad526cb83fe0495e6e6f1459516b760](https://github.com/magento/magento2/commit/08ecfe1efad526cb83fe0495e6e6f1459516b760)
 - [08daad82fdfde389be7695dcab5765de4f5326a1](https://github.com/magento/magento2/commit/08daad82fdfde389be7695dcab5765de4f5326a1)

The fix will be available with the upcoming `2.2.10` release.

**Comment by magento-engcom-team** [comment]
Hi @pushpendra07, @Nazar65, @krishprakash. 

Thank you for your report and collaboration!

The issue was fixed by Magento team. The fix was delivered into `magento/magento2:2.3-develop` branch(es).
Related commit(s):
 - [da0c7374e5dcae81735552bdd685ee5e646b8486](https://github.com/magento/magento2/commit/da0c7374e5dcae81735552bdd685ee5e646b8486)
 - [4344cc9bc34fbda80dc798ebafcf80c516788442](https://github.com/magento/magento2/commit/4344cc9bc34fbda80dc798ebafcf80c516788442)

The fix will be available with the upcoming `2.3.3` release.
